### PR TITLE
feat(vercel): Add basic deploy webhook support

### PR DESCRIPTION
**Pull request by MeredithAnya**
In order to work with release for the Vercel integration, we need to subscribe to the [webhook for deployments](https://vercel.com/docs/api?query=webh#integrations/webhooks/event-payloads/deployment). This PR covers 
* subscribing/creating to the webhooks during installation
* adding the webhook endpoint 
* validating the webhook signature in the endpoint

I've also added the `get_client` method on the integration to handle passing in the team_id to the `VercelClient` if necessary.

### CODE REVIEW

File: src/sentry/integrations/vercel/client.py [hunk 1]
```diff
@@ -35,3 +37,12 @@ def get_user(self):
     def get_projects(self):
         # TODO: we will need pagination since we are limited to 20
         return self.get(self.PROJECTS_URL)["projects"]
+
+    def create_deploy_webhook(self):
+        data = {
+            "name": "sentry_webhook",
```
Comment (MeredithAnya) [inline src/sentry/integrations/vercel/client.py:43]: @scefali don't have a strong attachment to this name, so we can change it if we want

Comment (scefali) [reply src/sentry/integrations/vercel/client.py:43]: @MeredithAnya Maybe "Sentry webhook"? The examples I see in their docs use spaces and capital letters.

File: src/sentry/integrations/vercel/integration.py [hunk 2]
```diff
@@ -115,13 +127,28 @@ def build_integration(self, state):
...
+            webhook = client.create_deploy_webhook()
+        except ApiError as err:
+            logger.info(
+                "vercel.create_webhook.failed",
+                extra={"error": six.text_type(err), "external_id": external_id},
+            )
+            try:
+                details = err.json["messages"][0].values().pop()
+            except Exception:
+                details = ""
```
Comment (scefali) [inline src/sentry/integrations/vercel/integration.py:140]: @MeredithAnya Nit: maybe in this case we should say something like `unknown error`

File: src/sentry/integrations/vercel/webhook.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+class VercelWebhookEndpoint(Endpoint):
+    authentication_classes = ()
+    permission_classes = ()
+
+    @csrf_exempt
+    def dispatch(self, request, *args, **kwargs):
+        return super(VercelWebhookEndpoint, self).dispatch(request, *args, **kwargs)
+
+    @transaction_start("VercelWebhookEndpoint")
```
Comment (scefali) [inline src/sentry/integrations/vercel/webhook.py:35]: @MeredithAnya Smart idea to use transactions here. I'll have to keep that in mind :)

File: tests/sentry/integrations/vercel/test_webhook.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+secret = "AiK52QASLJXmCXX3X9gO2Zyh"
+
+
+class VercelWebhookTest(APITestCase):
+    def setUp(self):
+        self.url = "/extensions/vercel/webhook/"
+
+    def test_get(self):
+        response = self.client.get(self.url)
+        assert response.status_code == 405
```
Comment (scefali) [inline tests/sentry/integrations/vercel/test_webhook.py:18]: @MeredithAnya Where does the `405` status code come from?

**Review by scefali — COMMENTED** [review]
Looks good overall! Just a few minor comments.

File: tests/sentry/integrations/vercel/test_webhook.py (hunk 4 above)
Comment (MeredithAnya) [reply tests/sentry/integrations/vercel/test_webhook.py:18]: @scefali I'm pretty sure it comes from the django `APIView` when you don't have the route defined. In other places we return `HttpResponse(status=405)` but  that's with using the generic `View`

Comment (scefali) [reply tests/sentry/integrations/vercel/test_webhook.py:18]: @MeredithAnya I see `405` means `Method not allowed`, so this test is basically saying you can't do a `GET` request on this route. Not sure why we need this test.

Comment (MeredithAnya) [reply tests/sentry/integrations/vercel/test_webhook.py:18]: I was just seeing that we test this for other webhook endpoints 🤷

**Review by scefali — APPROVED** [review]

**Pull request merged**